mono/metadata/reflection.c

```diff
@@ -459,6 +459,12 @@ mono_type_get_object_checked (MonoDomain *domain, MonoType *type, MonoError *err
 	 */
 	type = m_class_get_byval_arg (klass)->byref == type->byref ? m_class_get_byval_arg (klass) : m_class_get_this_arg (klass);
 
+	// We don't want to return types with custom modifiers to the managed
+	// world since they're hard to distinguish from plain types using the
+	// reflection APIs, but they are not ReferenceEqual to the unadorned
+	// types.
+	g_assert (!type->has_cmods);
```

Comment:
Is this a "temporarily we don't support this" or a "this will never work" scenario?

Reply:
It's a "this is a bug in the runtime if it ever happens" scenario.